### DISCUSSION TITLE
Can you stop an array query string having it's default values wiped if not all in query string?

### DISCUSSION THREAD

**Discussion by danjdewhurst** [livewire/livewire#5105] [1. Help]
Hi,

I've got a public property of `$filters` which is an array of default filter values, mainly null and empty string values.

It works on page load without any query parameters, however, it sets each filter value that isn't null straight away as I can't use the except functionality in an array (as far as I'm aware). This means that all my null filters get missed from the URL and then if the page reloads my public `$filters` property gets changed to only contain the empty string values.

I know I could get around this by re-hydrating my `$filters` if any keys are missing in the mount method, but this seems a little dirty.

Hope that all made sense,

Thanks

**Comment by joshhanley** [comment]
@danjdewhurst I don't think so, I think the whole array would need to be in the query string.

But you may be able to use the `except` keyword to do what you want, but you'd have to test it and see if it works how you would hope https://laravel-livewire.com/docs/2.x/query-string#clean-query-string

Hope this helps!
